example/why.js

```diff
@@ -1,7 +1,15 @@
 const PhishingDetector = require('../src/detector')
-let phishing = require('../src/config.json')
 
-const detector = new PhishingDetector(phishing)
+let phishing;
+let detector;
+
+fetch('https://raw.githubusercontent.com/MetaMask/eth-phishing-detect/main/src/config.json')
+  .then(response => response.json())
+  .then(data => {
+    phishing = data;
+    detector = new PhishingDetector(phishing);
+  })
+  .catch(error => console.error('Error:', error));
 
 function why (domain) {
   if (!detector) {
```

Comment:
It looks like previously, fetching the config was a synchronous action. Now it is asynchronous. However, `why` still thinks that it's not. Therefore when `why` is called, `detector` will be undefined and will return immediately. You'd have to call it again when the list is loaded for it to work. Is that the intent?

If not, what do you think about making `why` asynchronous and putting this logic in there instead? If you wanted to cache the result in case `why` was called again, then you could still keep a variable outside of the scope of `why` and consult it.

Perhaps something like this:

``` javascript
let phishing;
let detector;

async function why(domain) {
  if (!phishing || !detector) {
    const response = await fetch('https://raw.githubusercontent.com/MetaMask/eth-phishing-detect/main/src/config.json');
    const json = await response.json();
    phishing = data;
    detector = new PhishingDetector(phishing);
 }

 const reason = detector.check(domain);

 // ...
}
```

Now change `index.js` to:

``` javascript
window.addEventListener('load', function() {
  document.querySelector('form').addEventListener('submit', async (ev) => {
    ev.preventDefault()
    result.innerText = await why(input.value);
  });
});
```

(or if you like)

``` javascript
window.addEventListener('load', function() {
  document.querySelector('form').addEventListener('submit', (ev) => {
    ev.preventDefault()
    why(input).then((reason) => {
      result.innerText = reason;
    });
  });
});
```

Reply:
Yes, I knowingly made this compromise. This is a cheap hacky solution to a tool that otherwise is not serving its purpose. I'd love to tighten it up later, but I am stretched thin right now.